app/models/ability.rb

```diff
@@ -0,0 +1,13 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    user ||= User.new
+    p 'abilit ---', user
+    return unless user.present?
+
+    can :read, :all
+    can :manage, Post, author_id: user.id
+    can :delete, Post, author_id: user.id # if user.admin?
+  end
+end
```

Comment:
- [ **OPTIONAL** ] I think it would be better if you create a method named ``admin?`` inside the user model to for best practices and improve your code base syntax. Example
```ruby
...
   def admin?
     role == 'admin' ? true : false
   end
```